### DISCUSSION TITLE
`Binding to Task` can be run on AOT or Android?

### DISCUSSION THREAD

**Discussion by Coloryr** [AvaloniaUI/Avalonia#12735] [Q&A] [answered]
In Doc
```
https://docs.avaloniaui.net/zh-Hans/docs/next/guides/data-binding/how-to-bind-to-a-task-result
```

Run in windows not AOT
![image](https://github.com/AvaloniaUI/Avalonia/assets/26276037/3891dab4-9495-4e44-a4eb-4d8ef48c5423)
Is work

Run in windows with AOT

Run in Android
![Screenshot_2023-08-31-14-15-14-543_coloryr ColorM](https://github.com/AvaloniaUI/Avalonia/assets/26276037/925889fd-04fe-41fa-b748-8a2e9959e4ef)
No work

**Comment by Coloryr** [comment]
```
<Image
    Width="90"
    Height="90"
    HorizontalAlignment="Left"
    VerticalAlignment="Center"
    Source="{Binding Image^, FallbackValue={x:Static app:App.LoadIcon}}" />
```

```
public Task<Bitmap?> Image => GetImage();
private async Task<Bitmap?> GetImage()
{
    if (Data?.Logo == null)
    {
        return null;
    }
    try
    {
        _img = await ImageUtils.Load(Data.Logo);
        return _img;
    }
    catch (Exception e)
    {
        Logs.Error(App.GetLanguage("AddModPackWindow.Error5"), e);
    }

    return null;
}
```

**Reply by maxkatz6** [reply, marked as answer]
Is it with compiled bindings?

**Reply by Coloryr** [reply]
use compiled bindings work

**Comment by timunie** [comment]
@Coloryr do you have the permissions to open these files? Android is sandboxed, so any I/O operation needs permissions granted from the user.
